core/src/main/java/org/infinispan/factories/EmptyConstructorNamedCacheFactory.java

```diff
@@ -116,7 +117,12 @@
          } else if (componentType.equals(EvictionManager.class)) {
             return (T) new EvictionManagerImpl();
          } else if (componentType.equals(ExpirationManager.class)) {
-            return (T) new ExpirationManagerImpl();
+            CacheMode cacheMode = configuration.clustering().cacheMode();
+            if (cacheMode.isDistributed() || cacheMode.isReplicated()) {
```

Comment:
Yeah I actually had one written before but I noticed other guys in this factory were already doing this.  I can add a new one though.